src/ProtocolV3TestBase.sol

```diff
@@ -1535,6 +1538,29 @@ contract ProtocolV3TestBase is CommonTestBase {
       oracle.getSourceOfAsset(asset) == expectedSource,
       '_validateAssetSourceOnOracle() : INVALID_PRICE_SOURCE'
     );
+
+    require(
+      IOracleLike(oracle.getSourceOfAsset(asset)).decimals() == 8,
+      '_validateAssetSourceOnOracle() : INVALID_PRICE_SOURCE_DECIMALS'
+    );
+  }
+
+  function _validateAssetSourceLatestAnswerOnOracle(
+    IPoolAddressesProvider addressesProvider,
+    address asset,
+    address expectedSource
+  ) internal view {
+    IAaveOracle oracle = IAaveOracle(addressesProvider.getPriceOracle());
+
+    require(
+      IOracleLike(oracle.getSourceOfAsset(asset)).latestAnswer() >= 1_00000000,
+      '_validateAssetSourceOnOracle() : INVALID_PRICE_TOO_LOW'
+    );
+
+    require(
+      IOracleLike(oracle.getSourceOfAsset(asset)).latestAnswer() <= 1_000_000_00000000,
+      '_validateAssetSourceOnOracle() : INVALID_PRICE_TOO_HIGH'
+    );
```

Comment:
This is only checked when an oracle is changed. I would prefer to just check this on all assets under a new test.

Ex) `test_oracles_valid` or something which just loops over the market before and after the spell executes to make sure all oracles always follow proper format.

Reply:
Yeah this is what I thought as well